CI run against Twisted trunk is failing

See https://github.com/matrix-org/synapse/actions/runs/2500711697

```
tests/state/test_v2.py:512: error: Argument 1 to "ensureDeferred" has incompatible type "Coroutine[Any, Any, Mapping[Tuple[str, str], str]]"; expected "Union[Coroutine[Deferred[Dict[Tuple[str, str], str]], Any, Dict[Tuple[str, str], str]], Generator[Deferred[Dict[Tuple[str, str], str]], Any, Dict[Tuple[str, str], str]], Deferred[Dict[Tuple[str, str], str]]]"  [arg-type]
tests/handlers/test_federation.py:122: error: Need type annotation for "sg"  [var-annotated]
tests/handlers/test_federation.py:123: error: Argument 1 to "successResultOf" of "_Assertions" has incompatible type "Coroutine[Any, Any, Optional[int]]"; expected "Deferred[<nothing>]"  [arg-type]
tests/handlers/test_federation.py:152: error: Need type annotation for "sg2"  [var-annotated]
tests/handlers/test_federation.py:152: error: Argument 1 to "successResultOf" of "_Assertions" has incompatible type "Coroutine[Any, Any, Optional[int]]"; expected "Deferred[<nothing>]"  [arg-type]
tests/handlers/test_federation.py:175: error: Need type annotation for "sg"  [var-annotated]
tests/handlers/test_federation.py:176: error: Argument 1 to "successResultOf" of "_Assertions" has incompatible type "Coroutine[Any, Any, Optional[int]]"; expected "Deferred[<nothing>]"  [arg-type]
tests/handlers/test_federation.py:206: error: Need type annotation for "sg2"  [var-annotated]
tests/handlers/test_federation.py:206: error: Argument 1 to "successResultOf" of "_Assertions" has incompatible type "Coroutine[Any, Any, Optional[int]]"; expected "Deferred[<nothing>]"  [arg-type]
```

Would guess Twisted changed type annotations recently. https://github.com/twisted/twisted/pull/1736 Probably?

--- a/changelog.d/13061.misc
+++ b/changelog.d/13061.misc
@@ -1,0 +1,1 @@
+Fix type checking errors against Twisted trunk.

--- a/tests/handlers/test_federation.py
+++ b/tests/handlers/test_federation.py
@@ -119,7 +119,7 @@ class FederationTestCase(unittest.FederatingHomeserverTestCase):
         join_event = self._build_and_send_join_event(OTHER_SERVER, OTHER_USER, room_id)
 
         # check the state group
-        sg = self.successResultOf(
+        sg = self.get_success(
             self.store._get_state_group_for_event(join_event.event_id)
         )
 
@@ -149,7 +149,7 @@ class FederationTestCase(unittest.FederatingHomeserverTestCase):
         self.assertIsNotNone(e.rejected_reason)
 
         # ... and the state group should be the same as before
-        sg2 = self.successResultOf(self.store._get_state_group_for_event(ev.event_id))
+        sg2 = self.get_success(self.store._get_state_group_for_event(ev.event_id))
 
         self.assertEqual(sg, sg2)
 
@@ -172,7 +172,7 @@ class FederationTestCase(unittest.FederatingHomeserverTestCase):
         join_event = self._build_and_send_join_event(OTHER_SERVER, OTHER_USER, room_id)
 
         # check the state group
-        sg = self.successResultOf(
+        sg = self.get_success(
             self.store._get_state_group_for_event(join_event.event_id)
         )
 
@@ -203,7 +203,7 @@ class FederationTestCase(unittest.FederatingHomeserverTestCase):
         self.assertIsNotNone(e.rejected_reason)
 
         # ... and the state group should be the same as before
-        sg2 = self.successResultOf(self.store._get_state_group_for_event(ev.event_id))
+        sg2 = self.get_success(self.store._get_state_group_for_event(ev.event_id))
 
         self.assertEqual(sg, sg2)
 

--- a/tests/state/test_v2.py
+++ b/tests/state/test_v2.py
@@ -495,6 +495,7 @@ class StateTestCase(unittest.TestCase):
 
             prev_events = list(graph[node_id])
 
+            state_before: StateMap[str]
             if len(prev_events) == 0:
                 state_before = {}
             elif len(prev_events) == 1:

